utils/users.js

```diff
@@ -218,6 +221,56 @@ const parseSearchQuery = (queryString) => {
   return searchParams;
 };
 
+/**
+ * Generates discord nickname for a user
+ *
+ * @param {string} username - The discord username of the user.
+ * @returns {string} - Nickname of the user.
+ */
+const generateOOONickname = (username = "", from, until) => {
```

Comment:
here  the generateOOONickname() function. The function uses the substring() method to truncate the username to fit within the Discord nickname length limit. However, the substring() method does not take into account the length of the OOO message. This can result in the username being truncated too short, and the OOO message being truncated or omitted entirely.

Reply:
Yes, this was discussed with before with Ankush during discovery. It was decided that when a user is in OOO, the OOO dates should be given priority even if that results in the username being temporarily shortened. Hence the username (and thus it's length) in the nickname is decided on the basis of the length of OOO message. 
```
 const nicknameLen = discordNicknameLength - oooMessage.length - 1;
 return `${username.substring(0, nicknameLen)} ${oooMessage}`;
  ```